process-manager.js

```diff
@@ -42,19 +42,25 @@ class ProcessWrapper extends EventEmitter {
 	}
 }
 
+// execFile - the path to the file to spawn the child process(es) from
+// maxProcesses - the maximum number of child processes to spawn
+// isChatBased - the process manager handles some chat functionality
 class ProcessManager {
 	constructor(options) {
-		if (!options) options = {};
+		if (!('execFile' in options) || !('maxProcesses' in options) || !('isChatBased' in options)) {
+			Monitor.debug(
+				"An options object given to the ProcessManager constructor is " +
+				"missing required keys! The filename given is: " + (options.execFile || '""') + "."
+			);
+		}
 
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = options.execFile || '';
```

Comment:
It works by not working at all! I think I was going for making it a string by default, but that makes no sense either so I probably wrote this after staying up too long or something. Errors would actually make sense